x-pack/plugins/infra/server/services/rules/rule_data_client.ts

```diff
@@ -25,51 +23,23 @@ export const createRuleDataClient = ({
   logger: Logger;
   ruleDataService: RuleRegistryPluginSetupContract['ruleDataService'];
 }) => {
-  const initializeRuleDataTemplates = once(async () => {
-    const componentTemplateName = ruleDataService.getFullAssetName(
-      `${registrationContext}-mappings`
-    );
-
-    const indexNamePattern = ruleDataService.getFullAssetName(`${registrationContext}*`);
-
-    if (!ruleDataService.isWriteEnabled()) {
-      return;
-    }
-
-    await ruleDataService.createOrUpdateComponentTemplate({
-      name: componentTemplateName,
-      body: {
-        template: {
-          settings: {
-            number_of_shards: 1,
-          },
-          mappings: {},
+  return ruleDataService.initializeIndex({
+    feature: ownerFeatureId,
+    registrationContext,
+    dataset: Dataset.alerts,
+    componentTemplateRefs: [],
+    componentTemplates: [
+      {
+        name: 'mappings',
+        version: 0,
```

Comment:
Proper versioning according to the recent discussions will be implemented in a follow-up PR.